### PR TITLE
Solve luasec integration issues

### PR DESCRIPTION
**Pull request by rnikhil275**

### CODE REVIEW

**Comment by rnikhil275** [comment]
Like suggested [here](https://github.com/brunoos/luasec/pull/38#issuecomment-87273090) by @Tieske  in the comment, I included the entire http.lua file from luasocket with some changes.

**Comment by daurnimator** [comment]
'http2' is really terrible naming here: it suggests HTTP2 protocol support.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 /src/*.o
 /src/luasocket/*.o
 /*.dll
+/samples/certs/
```
Comment (daurnimator) [inline .gitignore]: missing newline at end of file

**Comment by rnikhil275** [comment]
@daurnimator HTTP2 support would be added once the integration issues(redirects, proxy etc) are solved. I would be working on this the entire summer. You can see the actual roadmap [here](https://docs.google.com/document/d/1Fsl_mXGRHeCxcqYecj5P7bYeRP0ipD9w0_CBnfa6fgU/edit#bookmark=id.9u8rt6t6m52i)

**Comment by daurnimator** [comment]
> @daurnimator HTTP2 support would be added once the integration issues(redirects, proxy etc) are solved. I would be working on this the entire summer. You can see the actual roadmap here

Oh.. huh... I already wrote a lua http library with http 2 support here: https://github.com/daurnimator/lua-http/. and it includes all the features you're adding....

**Comment by rnikhil275** [comment]
A pull request shall be opened once the work is complete

**Pull request closed without merging**